KarhooSDK/Service/User/KarhooPaymentProviderUpdateHandler.swift

```diff
@@ -0,0 +1,59 @@
+//
+// Created by Bartlomiej Sopala on 18/07/2022.
+//
+
+import Foundation
+
+class KarhooPaymentProviderUpdateHandler: PaymentProviderUpdateHandler {
+
+    private let userDataStore: UserDataStore
+    private let nonceRequestSender: RequestSender
+    private let paymentProviderRequest: RequestSender
+    private let loyaltyProviderRequest: RequestSender
+
+    init(
+        userDataStore: UserDataStore = DefaultUserDataStore(),
+        nonceRequestSender: RequestSender,
+        paymentProviderRequest: RequestSender,
+        loyaltyProviderRequest: RequestSender = KarhooRequestSender(httpClient: JsonHttpClient.shared)
```

Comment:
This needs to follow the same principle as the other two. Sometimes it's JsonHttpClient, but it can also be TokenRefreshingHttpClient